### PR TITLE
Add support for QSFP-DD cables on MLNX platform API

### PR DESCRIPTION
**Pull request by shlomibitton**
**- What I did**
Add support for QSFP-DD cables on MLNX platform API.

**- How I did it**
Used the new parser and EEPROM addresses for the new cable type to update results on 'xcvrd' daemon.

**- How to verify it**
Check for Transceiver DB data with QSFP-DD cable.
Check 'show interfaces transceiver eeprom Ethernet#' output.

**- Previous command output (if the output of a command-line utility has changed)**
**- New command output (if the output of a command-line utility has changed)**

### CODE REVIEW

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py [hunk 1]
```diff
@@ -513,46 +617,162 @@ def get_transceiver_info(self):
                 return None
 
             transceiver_info_dict['type'] = sfp_type_data['data']['type']['value']
-            transceiver_info_dict['manufacturename'] = sfp_vendor_name_data['data']['Vendor Name']['value']
-            transceiver_info_dict['modelname'] = sfp_vendor_pn_data['data']['Vendor PN']['value']
-            transceiver_info_dict['hardwarerev'] = sfp_vendor_rev_data['data']['Vendor Rev']['value']
-            transceiver_info_dict['serialnum'] = sfp_vendor_sn_data['data']['Vendor SN']['value']
+            transceiver_info_dict['manufacturer'] = sfp_vendor_name_data['data']['Vendor Name']['value']
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py:620]: maybe your branch haven't included Joe change to these field names?

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py [hunk 2]
```diff
@@ -466,7 +569,8 @@ def get_transceiver_info(self):
         specification_compliance   |1*255VCHAR     |specification compliance
         vendor_date                |1*255VCHAR     |vendor date
         vendor_oui                 |1*255VCHAR     |vendor OUI
-        ========================================================================
+        application_advertisement  |1*255VCHAR     |supported applications advertisement
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py:572]: the function definition in base class need to be updated.

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py [hunk 3]
```diff
@@ -719,6 +940,90 @@ def get_transceiver_bulk_status(self):
             transceiver_dom_info_dict['tx3bias'] = dom_channel_monitor_data['data']['TX3Bias']['value']
             transceiver_dom_info_dict['tx4bias'] = dom_channel_monitor_data['data']['TX4Bias']['value']
 
+        elif self.sfp_type == QSFP_DD_TYPE:
+            transceiver_dom_info_dict = {}
+
+            dom_info_dict_keys = ['temperature',    'voltage',
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py:946]: will different lengths of dicts for different types of cable causing issue in CLI part?

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py (hunk 1 above)
Comment (shlomibitton) [reply platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py:620]: Joe change to these field is on:
* sonic-platform-common
* sonic-platform-daemons
this is to adapt to the new names on API side, sonic-buildimage repo

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py (hunk 2 above)
Comment (shlomibitton) [reply platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py:572]: Added. It is located on 'sonic-platform-common', can be found in this PR:
https://github.com/shlomibitton/sonic-platform-common/pull/1/files

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py (hunk 3 above)
Comment (shlomibitton) [reply platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py]: It is not causing issue on CLI part, each entry with value of 'N/A' will be ignored on CLI.
You are correct, the current implementation is to return, for example for SFP cable, lane 1 value and lanes 2-4 with 'N\A'.
It means with this cable type is the same with 4 more lanes. I have updated the current implementation to be generic to all types in the same way it was until now with additional 4 'N/A' lanes.

**Pull request closed without merging**